### PR TITLE
theia-full: update image to include ext extensions

### LINKED ISSUE #65: [eslint] add 'eslint' support for ts, js images
Would be very helpful for JavaScript development to have inline eslint warnings in Theia, along the lines of what we currently have for Typescript.

FWIW, this is something currently supported by https://codesandbox.io/

### LINKED ISSUE #285: [Java, full] Language Server not working/starting
**Description**
I created a container from theia image and created a java file. But, auto suggestion is not working.

**Error**
Failed to start java language server.

Screenshot_2020-01-17 project — Theia Multi-Language Example

### PR DESCRIPTION
**Pull request by vince-fugnitto**
Fixes #285
Fixes #65 

#### What it does

The following pull-request updates the `theia-full` image (both `latest` and `next`) to include both vscode builtin extensions, and the necessary external extensions:
- updates `theia-full:latest` to use builtin and external extensions
- updates `theia-full:next` to use builtin and external extensions
- removes deprecated extensions from `theia-full:latest`
- removes deprecated extensions from `theia-full:next`
- uses the `download:plugins` script to download extensions from GitHub releases (as `.vsix` files)

#### How to test

Verify that the CI successfully passes (`theia-full`)

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: Vincent Fugnitto <vincent.fugnitto@ericsson.com>

### CODE REVIEW

**Comment by marcdumais-work** [comment]
Fixes https://github.com/theia-ide/theia-apps/issues/285

**Comment by vince-fugnitto** [comment]
> Fixes #285

I updated the description to reflect that it fixes #285 and #65 :)

**Comment by marcdumais-work** [comment]
@vince-fugnitto Did you intend to add the `eslint` extension? I do not see it.

**Comment by marcdumais-work** [comment]
**For Python:**

Using the same [tutorial](https://github.com/Microsoft/vscode-docs/blob/master/docs/python/tutorial-flask.md) I used for the Python-specific image, I found a few rough corners:
- I needed to set a couple of environment variables for the program to run (suggested in error message I got):
```
export LC_ALL=C.UTF-8
export LANG=C.UTF-8
```
I guess we should set these in `Dockerfile`, towards the end with other environment variables? e.g.:

```
ENV LC_ALL = C.UTF-8
ENV LANG = C.UTF-8
```

- When the extension detects I do not have a linter installed it proposes to install `pylint`. I accept and the following error is outputted:
```
(env) theia@cafed3629381:/home/project$ /home/project/env/bin/python -m pip install -U pylint
Collecting pylint
  Downloading https://files.pythonhosted.org/packages/e9/59/43fc36c5ee316bb9aeb7cf5329cdbdca89e5749c34d5602753827c0aa2dc/pylint-2.4.4-py3-none-any.whl (302kB)
    100% |################################| 307kB 2.4MB/s 
Collecting astroid<2.4,>=2.3.0 (from pylint)
  Downloading https://files.pythonhosted.org/packages/ad/ae/86734823047962e7b8c8529186a1ac4a7ca19aaf1aa0c7713c022ef593fd/astroid-2.3.3-py3-none-any.whl (205kB)
    100% |################################| 215kB 2.7MB/s 
Collecting isort<5,>=4.2.5 (from pylint)
  Downloading https://files.pythonhosted.org/packages/e5/b0/c121fd1fa3419ea9bfd55c7f9c4fedfec5143208d8c7ad3ce3db6c623c21/isort-4.3.21-py2.py3-none-any.whl (42kB)
    100% |################################| 51kB 7.0MB/s 
Collecting mccabe<0.7,>=0.6 (from pylint)
  Downloading https://files.pythonhosted.org/packages/87/89/479dc97e18549e21354893e4ee4ef36db1d237534982482c3681ee6e7b57/mccabe-0.6.1-py2.py3-none-any.whl
Collecting typed-ast<1.5,>=1.4.0; implementation_name == "cpython" and python_version < "3.8" (from astroid<2.4,>=2.3.0->pylint)
  Downloading https://files.pythonhosted.org/packages/90/ed/5459080d95eb87a02fe860d447197be63b6e2b5e9ff73c2b0a85622994f4/typed_ast-1.4.1-cp36-cp36m-manylinux1_x86_64.whl (737kB)
    100% |################################| 747kB 1.6MB/s 
Collecting wrapt==1.11.* (from astroid<2.4,>=2.3.0->pylint)
  Downloading https://files.pythonhosted.org/packages/23/84/323c2415280bc4fc880ac5050dddfb3c8062c2552b34c2e512eb4aa68f79/wrapt-1.11.2.tar.gz
Collecting lazy-object-proxy==1.4.* (from astroid<2.4,>=2.3.0->pylint)
  Downloading https://files.pythonhosted.org/packages/0b/dd/b1e3407e9e6913cf178e506cd0dee818e58694d9a5cd1984e3f6a8b9a10f/lazy_object_proxy-1.4.3-cp36-cp36m-manylinux1_x86_64.whl (55kB)
    100% |################################| 61kB 6.2MB/s 
Collecting six~=1.12 (from astroid<2.4,>=2.3.0->pylint)
  Downloading https://files.pythonhosted.org/packages/65/eb/1f97cb97bfc2390a276969c6fae16075da282f5058082d4cb10c6c5c1dba/six-1.14.0-py2.py3-none-any.whl
Building wheels for collected packages: wrapt
  Running setup.py bdist_wheel for wrapt ... error
  Complete output from command /home/project/env/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-zvto4gnu/wrapt/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmpuehs8gkjpip-wheel- --python-tag cp36:
  usage: -c [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
     or: -c --help [cmd1 cmd2 ...]
     or: -c --help-commands
     or: -c cmd --help
  
  error: invalid command 'bdist_wheel'
  
  ----------------------------------------
  Failed building wheel for wrapt
  Running setup.py clean for wrapt
Failed to build wrapt
Installing collected packages: typed-ast, wrapt, lazy-object-proxy, six, astroid, isort, mccabe, pylint
  Running setup.py install for wrapt ... done
Successfully installed astroid-2.3.3 isort-4.3.21 lazy-object-proxy-1.4.3 mccabe-0.6.1 pylint-2.4.4 six-1.14.0 typed-ast-1.4.1 wrapt-1.11.2
(env) theia@cafed3629381:/home/project$ 
```

Despite the error, I confirmed `pylint` reports linting errors so it seems functional in the end (I had to close and re-open the `.py` file for it to kick-in).

**Comment by marcdumais-work** [comment]
**Java**

Seems to work fine. I used the instructions [here](http://maven.apache.org/guides/getting-started/maven-in-five-minutes.html) to create a sample `maven` project. I was able to run, debug and run the project's tests successfully. 

:+1:

**Comment by marcdumais-work** [comment]
I did minimal sanity test for Go and Ruby (we only cover grammar) - looks good.

File: theia-full-docker/Dockerfile [hunk 1]
```diff
@@ -180,12 +184,15 @@ ADD $version.package.json ./package.json
...
-    NODE_OPTIONS="--max_old_space_size=4096" yarn theia build
+    NODE_OPTIONS="--max_old_space_size=4096" yarn theia build ;\
+    yarn theia download:plugins
 EXPOSE 3000
 ENV SHELL=/bin/bash \
     THEIA_DEFAULT_PLUGINS=local-dir:/home/theia/plugins
 
 ENV GOPATH=/home/project
 ENV PATH=$PATH:$GOPATH/bin
+ENV LC_ALL = C.UTF-8
```
Comment (marcdumais-work) [inline theia-full-docker/Dockerfile]: I think the spaces on either side of `=` are not good - the equal becomes part of the value of the env var:

```
(env) theia@ec807618bd38:/home/project$ echo $LC_ALL
= C.UTF-8

```

File: theia-full-docker/Dockerfile [hunk 2]
```diff
@@ -180,12 +184,15 @@ ADD $version.package.json ./package.json
...
+ENV LANG = C.UTF-8
```
Comment (marcdumais-work) [inline theia-full-docker/Dockerfile]: same

**Comment by vince-fugnitto** [comment]
> Retested Python and with locale set, now works OOTB - LGTM, thanks @vince-fugnitto .

Thank you for the review @marcdumais-work!

**Pull request merged**